openassessment/assessment/api/staff.py

```diff
@@ -0,0 +1,292 @@
+"""
+Public interface for staff grading, used by students/course staff.
+"""
+import logging
+from django.db import DatabaseError, IntegrityError, transaction
+from dogapi import dog_stats_api
+
+from openassessment.assessment.models import (
+    Assessment, AssessmentFeedback, AssessmentPart,
+    InvalidRubricSelection
+)
+from openassessment.assessment.serializers import (
+    AssessmentFeedbackSerializer, RubricSerializer,
+    full_assessment_dict, rubric_from_dict, serialize_assessments,
+    InvalidRubric
+)
+from openassessment.assessment.errors import (
+    StaffAssessmentRequestError, StaffAssessmentInternalError
+)
+from submissions import api as sub_api
+
+logger = logging.getLogger("openassessment.assessment.api.staff")
+
+
+STAFF_TYPE = "ST"
+
+
+def submitter_is_finished(submission_uuid, requirements):
+    """
+    Determine if the submitter has finished their requirements for staff
+    assessment. Always returns True.
+
+    Args:
+        submission_uuid (str): Not used.
+        requirements (dict): Not used.
+
+    Returns:
+        True
+
+    """
+    return True
+
+
+def assessment_is_finished(submission_uuid, requirements):
+    """
+    Determine if the assessment of the given submission is completed. This
+    checks to see if staff have completed the assessment.
+
+    Args:
+        submission_uuid (str): The UUID of the submission being graded.
+        requirements (dict): Any variables that may effect this state.
+
+    Returns:
+        True if the assessment has been completed for this submission.
+
+    """
+    if requirements and requirements.get('staff', {}).get('required', False):
+        return bool(get_latest_assessment(submission_uuid))
+    return True
+
+
+def get_score(submission_uuid, requirements):
+    """
+    Generate a score based on a completed assessment for the given submission.
+    If no assessment has been completed for this submission, this will return
+    None.
+
+    Args:
+        submission_uuid (str): The UUID for the submission to get a score for.
+        requirements (dict): Not used.
+
+    Returns:
+        A dictionary with the points earned, points possible,
+        contributing_assessments, and staff_id information.
+
+    """
+    assessment = get_latest_assessment(submission_uuid)
+    if not assessment:
+        return None
+
+    return {
+        "points_earned": assessment["points_earned"],
+        "points_possible": assessment["points_possible"],
+        "contributing_assessments": [assessment["id"]],
+        "staff_id": assessment["scorer_id"],
+    }
+
+
+def get_latest_assessment(submission_uuid):
+    """
+    Retrieve the latest staff assessment for a submission.
+
+    Args:
+        submission_uuid (str): The UUID of the submission being assessed.
+
+    Returns:
+        dict: The serialized assessment model
+        or None if no assessments are available
+
+    Raises:
+        StaffAssessmentInternalError
```

Comment:
If no submissions or assessments are found, it will return `None` and not error out. That error is caught for issues with contacting the database in the first place. I'll update the comment to make that clearer.